### ISSUE TITLE
RequestBlocksByRange block skipping

### ISSUE THREAD

**Issue by mpetrunic** [ChainSafe/lodestar#521]
Currently we ignore step flag when returning blocks. We should use given step number and skip blocks depending on step amount.

**Comment by aunyks** [comment]
Say we have 4 blocks with height 0, 1, 2, and 3, respectively, and are requested with a step of 2.  

Do we return 0 and 2, or 1 and 3? Or, is the intended behavior different from what I'm thinking?

**Comment by wemeetagain** [comment]
I think we would be returning 0 and 2 in that case.

**Comment by aunyks** [comment]
Okay, and if the step is 5 do we just return 0 and stay quiet, or do we throw an error and not return anything?

**Comment by wemeetagain** [comment]
In that case, just return 0.

**Comment by mpetrunic** [comment]
tldr depends on start slot, first slot is inclusive, last slot exclusive
![image](https://user-images.githubusercontent.com/8836210/70481893-48889e00-1ae4-11ea-9de9-bc1e279d5388.png)

**Comment by aunyks** [comment]
Hm, that sounds kinda tricky at first. Are there any other examples or discussions that could help me understand? Thanks for the screenshot btw
